src/Jackett.Common/Indexers/Abstract/AvistazTracker.cs

```diff
@@ -163,5 +179,27 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
             }
             return releases;
         }
+
+        private async Task<string> GetMovieId(string imdbId)
+        {
+            try
+            {
+                var imdbUrl = IMDBSearch + imdbId;
+                var imdbHeaders = new Dictionary<string, string> { {"X-Requested-With", "XMLHttpRequest"} };
+                var imdbResponse = await RequestStringWithCookiesAndRetry(imdbUrl, null, null, imdbHeaders);
+                if (imdbResponse.IsRedirect)
+                {
+                    // re-login
+                    await ApplyConfiguration(null);
+                    imdbResponse = await RequestStringWithCookiesAndRetry(imdbUrl, null, null, imdbHeaders);
+                }
+                var json = JsonConvert.DeserializeObject<dynamic>(imdbResponse.Content);
+                return (string)((JArray)json["data"])[0]["id"];
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
```

Comment:
This function feels like it should go somewhere outside the tracker for others to use.

Reply:
This function is used only by this tracker.